### PR TITLE
LPS-192003 | Automated test for CP2.0 |  AssertComplimentaryPortalKeyCanBeCreated

### PR DESCRIPTION
**Pull request by styller1**
Ticket:
[LPS-192003](https://liferay.atlassian.net/browse/LPS-192003)

### CODE REVIEW

**Comment by styller1** [comment]
ci:test:sf

**Comment by styller1** [comment]
ci:test:sf

**Comment by styller1** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase [hunk 1]
```diff
@@ -163,6 +135,98 @@ definition {
...
+
+				CP2NavigationMenu.clickOnProductButton(product = ${testProduct});
+			}
+		}
+
+		task ("And go to generate a new key") {
+			CP2ActivationKeys.generateNewKey();
+		}
+
+		task ("When fulfill first form for Generate New License") {
```
Comment (heiditrindade) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Change to "When filling in the first form to Generate New License"

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase [hunk 2]
```diff
@@ -163,6 +135,98 @@ definition {
...
+			CP2ActivationKeys.selectFistActiveSubscriptionInput(index = 3);
+
+			CustomerPortal.clickAtButton(text = "Next");
+
+			Click.clickAtNotVisible(locator1 = "CP2ActivationKeys#EXPIRATION_CHECKBOX");
+
+			CustomerPortal.clickAtButton(text = "Next");
+		}
+
+		task ("And when fulfill the Environment and Host name fields") {
```
Comment (heiditrindade) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Change to "And filling in the Environment and Host name fields"

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase [hunk 3]
```diff
@@ -163,6 +135,98 @@ definition {
...
+		}
+		
+
+		task("Then the key appears on table") {
+			CP2Utils.searchForKeyInTable(environmentName = "Environment Test Name");
+			
+			AssertElementPresent(
+			key_text = "Environment Test Name",
+			locator1 = "CP2ActivationKeys#RESPONSIVE_TABLE");
+		}
```
[the remark this answers is not in the document]
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Good suggestion, I will change

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase (hunk 2 above)
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Good suggestion, I will change

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase (hunk 1 above)
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Good suggestion, I will change

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase [hunk 4]
```diff
@@ -163,6 +135,98 @@ definition {
 		}
 	}
 
+	@description = "Automation Ticket: LPS-151337 | Test Summary: Verify that after filling out all the fields the Next button is enabled."
+	@ignore = "true"
+	@priority = 4
+	test AssertComplimentaryPortalKeyCanBeCreated {
+		
+		task ("Given Active the option in koroneiki 'Allow Self Provissioning'") {
```
[the remark this answers is not in the document]
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Good suggestion, I will change

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase [hunk 5]
```diff
@@ -163,6 +135,98 @@ definition {
...
+
+		task ("And the user Role is Changed") {
+			CP2Utils.assigneToRoleViaUsersTab(
+				searchItem = "test",
+				accountRoleTitle = "Administrator",
+				accountUserName = "Test Test",
+				testAccount = "2022 FIFA World Cup Qatar - FAN ID");
+		}
+
+		task ("And an user go to Customer Portal") {
```
[the remark this answers is not in the document]
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Good suggestion, I will change

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase [hunk 6]
```diff
@@ -163,6 +135,98 @@ definition {
 		}
 	}
 
+	@description = "Automation Ticket: LPS-151337 | Test Summary: Verify that after filling out all the fields the Next button is enabled."
```
[the remark this answers is not in the document]
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Good suggestion, I will change

[the remark this answers is not in the document]
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase (hunk 4 above)
[the remark this answers is not in the document]
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase (hunk 5 above)
[the remark this answers is not in the document]
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase (hunk 1 above)
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase (hunk 2 above)
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase (hunk 3 above)
[the remark this answers is not in the document]
Comment (styller1) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2NewKeyGenerationPortal.testcase]: Done!

**Comment by styller1** [comment]
ci:test:sf

**Comment by styller1** [comment]
Solved in [#4628](https://github.com/pt-liferay-solutions/liferay-portal/pull/4628)

**Pull request closed without merging**